src/frontends/tensorflow/src/graph_iterator_proto.hpp

```diff
@@ -88,29 +88,40 @@ class GraphIteratorProto : public GraphIterator {
         }
     }
 
-    /// Set iterator to the start position
+    /// \brief Check if the input file is supported
+    template <typename T>
+    static bool is_supported(const std::basic_string<T>& path) {
+        std::ifstream pb_stream(path, std::ios::in | std::ifstream::binary);
+        auto graph_def = std::make_shared<::tensorflow::GraphDef>();
+        return pb_stream && pb_stream.is_open() && graph_def->ParsePartialFromIstream(&pb_stream);
```

Comment:
Nervous place. But np if it isn't too heavy operation

Reply:
we have tests + newly added:)